swap_meet/vendor.py

```diff
@@ -1,2 +1,97 @@
-class Vendor:
-    pass
\ No newline at end of file
+from .item import Item
+
+class Vendor(Item):
+    def __init__(self, inventory = None):
+        if inventory is None:
+            inventory = []
+        self.inventory = inventory
+        
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        if item not in self.inventory:
+            return False
+
+        self.inventory.remove(item)
+        return item
+
+
+# ******** Wave 2 ********
+    def get_by_category(self,category = None):
+        if category is None:
+            category = ""
+        category_items = []
+        for item in self.inventory:
+            if item.category == category:
+                category_items.append(item)
+
+        return category_items
+
+
+# ******** Wave 3 ********
+    def swap_items(self, vendor, my_item, their_item):
+        if self.inventory == [] or vendor.inventory == []:
```

Comment:
This check isn't necessary! Since you have to check if the items are in the inventories, that takes care of the empty inventory check. An item can't exist in an empty list.